src/app/admin/openid-connect-provider/openid-connect-provider.component.ts

```diff
@@ -0,0 +1,129 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+import { Component, OnInit } from '@angular/core';
+import { Title } from '@angular/platform-browser';
+import { OpenIdConnectProviderDetail, OpenIdConnectProvidersDetailResponse, Uuid } from '@maurodatamapper/mdm-resources';
+import { MdmResourcesService } from '@mdm/modules/resources';
+import { MessageHandlerService, SharedService, StateHandlerService } from '@mdm/services';
+import { EditingService } from '@mdm/services/editing.service';
+import { UIRouterGlobals } from '@uirouter/angular';
+import { EMPTY, Observable } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { OpenIdConnectProviderForm } from './openid-connect-provider.model';
+
+@Component({
+  selector: 'mdm-openid-connect-provider',
+  templateUrl: './openid-connect-provider.component.html',
+  styleUrls: ['./openid-connect-provider.component.scss']
+})
+export class OpenidConnectProviderComponent implements OnInit {
+  id: Uuid;
+  editExisting = false;
+  form: OpenIdConnectProviderForm;
+  previewImageUrl?: string;
+
+  constructor(
+    private uiRouterGlobals: UIRouterGlobals,
+    private resources: MdmResourcesService,
+    private messageHandler: MessageHandlerService,
+    private stateHandler: StateHandlerService,
+    private editing: EditingService,
+    private shared: SharedService,
+    private title: Title) { }
+
+  ngOnInit(): void {
+    if (!this.shared.features.useOpenIdConnect) {
+      this.stateHandler.Go('alldatamodel');
+      return;
+    }
+
+    this.editing.start();
+
+    this.id = this.uiRouterGlobals.params.id;
+    this.editExisting = this.id !== undefined && this.id !== null;
+
+    if (this.editExisting) {
+      this.title.setTitle('OpenID Connect - Edit provider');
+      this.loadExistingProvider();
+    }
+    else {
+      this.title.setTitle('OpenID Connect - Add provider');
+      this.createForm();
+    }
+  }
+
+  refreshImagePreview() {
+    this.previewImageUrl = this.form.imageUrl.value;
+  }
+
+  cancel() {
+    this.editing.confirmCancelAsync().subscribe(confirm => {
+      if (confirm) {
+        this.navigateToParent();
+      }
+    });
+  }
+
+  save() {
+    if (this.form.group.invalid) {
+      return;
+    }
+
+    const payload = this.form.createPayload();
+    const request: Observable<OpenIdConnectProvidersDetailResponse> = this.editExisting
+      ? this.resources.pluginOpenIdConnect.update(this.id, payload)
+      : this.resources.pluginOpenIdConnect.save(payload);
+
+    request
+      .pipe(
+        catchError(error => {
+          this.messageHandler.showError('There was a problem saving the OpenID Connect provider.', error);
```

Comment:
I added some junk info and tried to save got the error message but also directed to a 500 screen, we might need to think about how we handle these

Reply:
Sounds like a wider issue. This error handling is consistent with other pages, so I think I'll leave this for now and assume backend won't throw these kinds of errors.